Archiving page with s3 options (#1176)
Graylog2/documentation: pages/archiving/setup.rst

when the server starts for the first time but you can create a new backend if you want to store the data in a different
path.

> S3 Archiving

Reviewer: ```suggestion
S3 Archiving Backend
```

~~~~~~~~~~~~

> Graylog provides archiving built to be used against AWS S3. This doesn't mean it's limited
> to the AWS solution. You can configure S3 compatible solutions like MinIO, CEPH, Digital Ocean 
> Spaces and others.

Reviewer: ```suggestion
The S3 Archiving backend can be used to upload archives to an AWS S3 object storage service.
It is built to work with AWS, but should be compatible with other S3 implementations like MinIO, CEPH, Digital Ocean 
Spaces, etc.
```

Author: Yeah it's a little more verbose, but I lean toward Oxford commas. Regardless, I like your suggestion. So I'll keep it. Thx

| Title                       | A simple title to identify the backend    |     
+-----------------------------+-------------------------------------------+
| Description                 | Longer description for the backend        |     
+-----------------------------+-------------------------------------------+                                                
| S3 Endpoint URL             | Only configure this if not using AWS      |                                                
+-----------------------------+-------------------------------------------+
| AWS Authentication Type     | Choose access type from the dropdown menu |     
+-----------------------------+-------------------------------------------+                                                 
| AWS Assume Role (ARN)       | This is an optional input for             |
|                             | alternate authentication mechanisms.      |      
+-----------------------------+-------------------------------------------+     
| Bucket Name                 | The name of the S3 bucket                 |     
+-----------------------------+-------------------------------------------+     

> | Spool Directory             | Temporary name for the archiving process  |    

Reviewer: Directory where archiving data is stored before being uploaded

Author: Updated

+-----------------------------+-------------------------------------------+     
| AWS Region                  | Choose *Automatic* or configure the       |
|                             | appropriate option                        |
+-----------------------------+-------------------------------------------+

> | S3 Output Base Path         | Creates archive structure                 |

Reviewer: Archives will be stored under that path

+-----------------------------+-------------------------------------------+

AWS Authentication Type
~~~~~~~~~~~~~~~~~~~~~~~

Graylog provides several options for granting access. You can:

* use the *Automatic* authentication mechanism if you provide AWS credentials through your file
  system or process environment.
* enter credentials manually

AWS Assume Role (ARN)
~~~~~~~~~~~~~~~~~~~~~

> This is typically used for buckets that you don’t own. For example, if someone gave you temporary 
> permissions to their bucket, you would need to use this configuration.

Reviewer: ```suggestion
This is typically used for allowing cross-account access to a bucket. 
See https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html 
for further details.
```

Spool Directory
~~~~~~~~~~~~~~~

> The Spool Directory is always writable for the upload process to persist.
> 
> Before Graylog upload archives to the S3, it has to write them into the file system. As soon as 
> Graylog writes these segments, they are uploaded. 
> 
> Keep in mind you need 5 gigabytes of free disk space in this directory to manage 10 segments at 
> 500 megabytes. However, it may use less memory because it deletes files as soon as they upload.

Reviewer: ```suggestion
The archiving process needs this directory to store some temporary data, before it can be uploaded to S3.

This directory should be writable and have enough space to fit 10 times the *Max Segment Size*  (add link to the config section)
```

AWS Region
~~~~~~~~~~

> By default, Graylog pulls the region from your file systems. 
> 
> If you choose to select a region, you should use the one where you created the bucket. 
> 
> If you are not using AWS, you do not need to configure this. 

Reviewer: ```suggestion
Select the AWS region where you archiving bucket resides in.
If no region is selected, Graylog tries to get the region from your file system or process environment.
If you are not using AWS, you do not need to configure this. 
```

+-----------------------------+-------------------------------------------+                                                     
| hour                        | Archival date hour                        |     
+-----------------------------+-------------------------------------------+     
| minute                      | Archival date minute                      |     
+-----------------------------+-------------------------------------------+     
| second                      | Archival date second                      |
+-----------------------------+-------------------------------------------+

Activate Backend
~~~~~~~~~~~~~~~~

After configuring your bucket, click *Save*.

This will bring you back to the *Edit archive backend configuration* page. 

To activate the backend, you need to:

#. Click on the *Configuration* tab located in the top righthand corner.
#. Under the *Backend* dropdown menu, select the backend you want to activate. 
#. You can choose to change configurations or use the defaults provided. 
#. Click the green *Update configuration* button at the bottom of the screen.
#. This will return you to the *Archives* screen.

> The archiving will run automatically.  
> 
> To review your S3 archive, click the *Archive Index* button. 

Reviewer: ```suggestion
```

I would not mention that here.
This isn't specific to the backend, and is probably covered in the Usage section
